src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
+    const order = orderByUtil.getValue();
+
+    Array.from(tBody.rows)
+      .sort(sortingFunctions.get(index, colHeader, order))
+      .forEach((row) => tBody.append(row));
+  }
+};
+
+tHead.addEventListener('click', handleSortingEvent);
+
+// HANDLING ROW SELECT EVENT
+const deselectRows = () => {
+  for (const row of table.rows) {
+    if (row.classList.contains('active')) {
+      row.classList.remove('active');
+    }
+  }
+};
+
+tBody.addEventListener('click', (e) => {
+  const target = e.target.closest('tr');
+
+  if (target) {
+    deselectRows();
+    target.classList.add('active');
+  }
+});
+
+// ADDING FORM TO THE DOCUMENT
+const formatLabelText = (inputName) => {
+  return `${inputName.charAt(0).toUpperCase()}${inputName.substring(1)}: `;
+};
+
+const createInput = (inputName, inputType) => {
+  const input = Object.assign(document.createElement('input'), {
+    name: inputName,
+    type: inputType,
+  });
+  const label = document.createElement('label');
+
+  input.setAttribute('data-qa', inputName);
+
+  label.textContent = formatLabelText(inputName);
+  label.append(input);
+
+  return label;
+};
+
+const createSelect = (inputName, options) => {
+  const select = Object.assign(document.createElement('select'), {
+    name: inputName,
+  });
+
+  select.setAttribute('data-qa', inputName);
+
+  Array.from(options)
+    .sort()
+    .forEach((option) => select.append(new Option(option, option)));
+
+  const label = document.createElement('label');
+
+  label.textContent = formatLabelText(inputName);
+  label.append(select);
+
+  return label;
+};
+
+const createButton = (btnText, type) => {
+  const button = document.createElement('button');
+
+  button.setAttribute('type', type);
+  button.textContent = btnText;
+
+  return button;
+};
+
+const form = document.createElement('form');
+const offices = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+const formElements = [
+  createInput('name', 'text'),
+  createInput('position', 'text'),
+  createSelect('office', offices),
+  createInput('age', 'number'),
+  createInput('salary', 'number'),
+  createButton('Save to table', 'submit'),
+];
+
+form.classList.add('new-employee-form');
+formElements.forEach((element) => form.append(element));
+document.addEventListener('DOMContentLoaded', () => document.body.append(form));
+
+// FORM VALIDATION
+const validateFormInputData = (value, key) => {
+  const validators = {
+    position: () =>
+      !value
+        ? { validated: false, message: `Position is a required field!` }
+        : { validated: true },
+    name: () =>
+      !value || value.length < 4
+        ? { validated: false, message: 'Name should be more than 4 chars long' }
+        : { validated: true },
+    age: () =>
+      value < 18 || value > 90
+        ? { validated: false, message: 'Age must be between 18 and 90' }
+        : { validated: true },
+  };
+
+  return validators[key]?.() || { validated: true };
+};
```

Comment:
The `validateFormInputData` function is a bit hard to read because of the nested functions. Consider simplifying this function or splitting it into smaller, more manageable functions.